autosklearn/evaluation/train_evaluator.py

```diff
@@ -979,10 +944,6 @@ def get_splitter(self, D: AbstractDataManager) -> Union[BaseCrossValidator, _Rep
                    D.info['task'] != MULTIOUTPUT_REGRESSION):
 
                     y = y.ravel()
-                if class_name == 'PredefinedSplit':
```

Comment:
Ha, this issue was badly named. I like where it is going, but I was actually hoping to remove the checks `if class_name == 'XXX'`. Do you think that's possible?

Reply:
The only reason why I still kept them is because of the group parameter requirement of some splitters.

scikit-learn will raise a value error like this:
https://github.com/scikit-learn/scikit-learn/blob/2beed55847ee70d363bdbfe14ee4401438fba057/sklearn/model_selection/_split.py#L969

I think we can add a check to the automl fit, so that we don't even start if groups is missing. To do this, we call the `get_n_splits(groups=groups) ` method. makes sense?

Reply:
That's a new scikit-learn method which didn't exist back then and seems to solve our issues here :)

Yes, could you please move to using that?

Reply:
I thought this resolved our problem. In the case of groupkfold it does not ask for a group argument, I have to do a `next()` on the splitter. I proposed a change with this but this maybe complicates the code a little? 

Let me know your thoughts...

Reply:
Would this be an issue to raise in the scikit-learn issue tracker? The change looks fine and is an overall improvement!